### ISSUE TITLE
Raspberry Pi 4 issues

### ISSUE THREAD

**Issue by alexruf** [ryansch/docker-unifi-rpi#54]
I tried to run the container on a Raspberry Pi 4. First it seemed like it would work, but the application is not coming up. I couldn't see anything helpful in the logs.
Does anyone know why it's not working on the new Raspberry Pi 4 and how this issues can be fixed?

**Comment by ryansch** [comment]
I just tested the controller on a pi 4 and it came up. Can you supply more information?

**Comment by alexruf** [comment]
Just seen I booted Raspbian with the [ARMv8 64bit Kernel](https://www.raspberrypi.org/documentation/configuration/config-txt/boot.md). It must have something to do with this since it's the only difference to the way my Raspberrry Pi 3 is configured.

I'll try switching back to the old 32bit Kernel. Would be too bad if just doesn't run with the new Kernel, then I wouldn't be able to make use of the 4 GB memory of the Raspberry Pi 4.

**Comment by alexruf** [comment]
> I'll try switching back to the old 32bit Kernel.

Interesting, it works just fine as soon as I switched back to the old ARMv7 32bit Kernel.

**Comment by ryansch** [comment]
@alexruf That _is_ strange as I'm building for `arm64/v8` as well. I'm running hypriot on mine at the moment so it's 32 bit only.

**Comment by ryansch** [comment]
It makes me wonder if docker's manifest check is wrong when the kernel and userspace are mismatched.

**Comment by florianschroen** [comment]
Same on rpi 3b+ with raspian buster!
when activating 64bit kernel via `/boot/config.txt`:
`arm_64bit=1`

```
$ uname -a
Linux cloudpi 4.19.79-v8+ #1273 SMP PREEMPT Fri Oct 11 18:35:44 BST 2019 aarch64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID: Raspbian
Description:    Raspbian GNU/Linux 10 (buster)
Release:        10
Codename:       buster

$ docker ps
CONTAINER ID        IMAGE                               COMMAND                  CREATED             STATUS              PORTS                                                                                                                                                                                                                              NAMES
e866307c71e6        ryansch/unifi-rpi:5.12.19-arm32v7   "/sbin/tini -g -- /d…"   6 days ago          Up 6 days           0.0.0.0:1900->1900/udp, 0.0.0.0:6789->6789/tcp, 0.0.0.0:8080->8080/tcp, 0.0.0.0:8443->8443/tcp, 0.0.0.0:8843->8843/tcp, 0.0.0.0:3478->3478/udp, 0.0.0.0:10001->10001/udp, 0.0.0.0:8880->8880/tcp, 8081/tcp, 0.0.0.0:80->8080/tcp   unifi

```
as you can see, the docker container is running since 6 days, but still no webinterface.

Using ryansch/unifi-rpi:5.12.19-arm64v8 is impossible. docker run command exits with return code 159.
I guess this is because it is impossible to start 64bit binaries out of a 32bit OS, even if using a 64bit kernel. Raspbian is 32bit only for simplicty.

While testing with 64bit, I mentioned using the "latest" tag is impossible at the moment. So i had to select a specific version tag. This nodered issue https://github.com/node-red/node-red-docker/issues/150 shows the same problem for rpi0w.


When switching back to 32bit by uncommenting `arm_64bit=1`, webinterface is starting without any issues.
```
$ uname -a
Linux cloudpi 4.19.79-v7+ #1273 SMP Fri Oct 11 18:13:16 BST 2019 armv7l GNU/Linux

$  docker ps
CONTAINER ID        IMAGE                               COMMAND                  CREATED             STATUS              PORTS                                                                                                                                                                                                                              NAMES
ab473599dbd0        ryansch/unifi-rpi:5.12.19-arm32v7   "/sbin/tini -g -- /d…"   2 minutes ago       Up 2 minutes        0.0.0.0:1900->1900/udp, 0.0.0.0:6789->6789/tcp, 0.0.0.0:8080->8080/tcp, 0.0.0.0:8443->8443/tcp, 0.0.0.0:8843->8843/tcp, 0.0.0.0:3478->3478/udp, 0.0.0.0:10001->10001/udp, 0.0.0.0:8880->8880/tcp, 8081/tcp, 0.0.0.0:80->8080/tcp   unifi
```

**Comment by ryansch** [comment]
Looks like a moby bug: https://github.com/moby/moby/issues/37647

**Comment by SvenDowideit** [comment]
gawd that's ancient - https://github.com/moby/moby/issues/34875#issuecomment-331171169

Phil suggests re-ordering the manifest for the multi-arch image is the only current solution :( (current - 2017!)

**Comment by ryansch** [comment]
@SvenDowideit I can give that a whirl.

**Comment by ryansch** [comment]
I finally circled around to this. I don't think this is the same issue as I'm not shipping support for multiple arm variants.

```sh
$ manifest-tool inspect ryansch/unifi-rpi:latest

Name:   ryansch/unifi-rpi:latest (Type: application/vnd.docker.distribution.manifest.list.v2+json)
Digest: sha256:576bd58cf5215f07171953414eaca39019a50928b218dc65426bdbb0670ea726
 * Contains 3 manifest references:
1    Mfst Type: application/vnd.docker.distribution.manifest.v2+json
1       Digest: sha256:df7a4a2d6eecef2f220d584ca9c9bf52506fd19e94a66592f21fb5ef8ad35a91
1  Mfst Length: 2208
1     Platform:
1           -      OS: linux
1           - OS Vers:
1           - OS Feat: []
1           -    Arch: amd64
1           - Variant:
1           - Feature:
1     # Layers: 9
         layer 1: digest = sha256:fc7181108d403205fda45b28dbddfa1cf07e772fa41244e44f53a341b8b1893d
         layer 2: digest = sha256:73f08ce352c86de44048828a8c20f22011f46efd4d03cab7269354f97b131688
         layer 3: digest = sha256:eac271a34b404ba767f634cb690410caaec27aa4e5f1b1bc1c8b73ef11f343c0
         layer 4: digest = sha256:7742200666120407ea150031f8deddbdc36132de2762a5ae072bf14fd256c803
         layer 5: digest = sha256:804cc32287eb0983b61812f6a3f0166a82c6c824a1c7fe421dd304165b45f5f8
         layer 6: digest = sha256:fd7f0a7f0aaf3d6b4c775aca6e627a8516d53917ebaf37cbed43db3a1da8b6c4
         layer 7: digest = sha256:411a88da1f5434e6b678f80971e10de6a0c561782023684048697efb181fc6b4
         layer 8: digest = sha256:2c178fd77bea5ff50244c5942e5fbae6881d74a84f69ad4fd1af0923f44679ea
         layer 9: digest = sha256:9441db386efd1db2cb14b114a3027b0744d5362c4c62b67c80b4f5358c1aadca

2    Mfst Type: application/vnd.docker.distribution.manifest.v2+json
2       Digest: sha256:9f1c012db46c0315642cf4a95b1114531f86bc6340bf6d01f3baf414523397b0
2  Mfst Length: 2624
2     Platform:
2           -      OS: linux
2           - OS Vers:
2           - OS Feat: []
2           -    Arch: arm
2           - Variant: v7
2           - Feature:
2     # Layers: 11
         layer 1: digest = sha256:b0727b1ec48ef8920082aec023c050c83b563ae08080d9951d68fba55934bbd6
         layer 2: digest = sha256:c1e4b3bd395a6982a3d52e8e2c57651fb045081876d3ca9e21f248e1d6bc74cb
         layer 3: digest = sha256:c3a541a7bc5211dd1a923061f4978b3cf7a6b4b3144dbdb679863fe09240886a
         layer 4: digest = sha256:9581944edf718a07e1853240b2d7caec780beaf162aec89a02314401ec2b62c6
         layer 5: digest = sha256:b3983e778085318897d7ea5368a2cedd24a2ce2fd0c33f66ac5628125acc5b13
         layer 6: digest = sha256:a3eed903012bb5bead8a4c5d0a6b9ec2282c720778f6c8ef2df2c0cf2e9a1ebb
         layer 7: digest = sha256:a4094100fdffb0796d86a66e827aee526bbcdd8a224555d3a76ee647120872ab
         layer 8: digest = sha256:050068d4a02bc3670683ce5710e46dfb234ade3291a3ed16528dca819c7a61bc
         layer 9: digest = sha256:7f89d94502192c120571af037f1ec5eade4a56914cb4c716cda848df9c8aebcd
         layer 10: digest = sha256:35305925e858d2f15266798904f6c27bace3c981565935de974dc47c574c1e40
         layer 11: digest = sha256:9441db386efd1db2cb14b114a3027b0744d5362c4c62b67c80b4f5358c1aadca

3    Mfst Type: application/vnd.docker.distribution.manifest.v2+json
3       Digest: sha256:d46b59714a2ce4b959b368a3d226588210813c72a7a0c95d7c3bfa366dd61a6e
3  Mfst Length: 2624
3     Platform:
3           -      OS: linux
3           - OS Vers:
3           - OS Feat: []
3           -    Arch: arm64
3           - Variant: v8
3           - Feature:
3     # Layers: 11
         layer 1: digest = sha256:29b80961214d7f0c89081fe8134e6e8e14ccfa1afe001357539f59930ff9e3ef
         layer 2: digest = sha256:392e0924564e298648b4a7b2ebc774103f98526dd2afd1c948697b4c6686300f
         layer 3: digest = sha256:d15c6e01e710610402034472504e3628bdd00c2b0907f3ec4364f8e71a38d8c0
         layer 4: digest = sha256:acba4335cd44174a82b7a5d29d7b9dacc9e0b19f066b622c972c573edeb4dddd
         layer 5: digest = sha256:b1f6a0f413ede3cb63ee3cc7b5fa3c65a3ef3c039ac50bb2399b194940c6fd2e
         layer 6: digest = sha256:3f724991c026fdfad29444b78aaaa27fa0489399eff887105d0b0dbb3c646d15
         layer 7: digest = sha256:1830b0590d6e909e75c25419af06f227cd60e069f037db09d4650d2723a283b0
         layer 8: digest = sha256:76b95d00172ce74ee06acdfc4d27d908ad3747616fb9724b3ffbb2132e9dc1e7
         layer 9: digest = sha256:0affba9a939f4e0215942833f1586bc5e68fde2c3c9db629892b5ddfb4974a26
         layer 10: digest = sha256:d669f8c7ce87e71d68b760ee29d62a22c55b02cf28e0da212dcd0edde228fd45
         layer 11: digest = sha256:9441db386efd1db2cb14b114a3027b0744d5362c4c62b67c80b4f5358c1aadca
```

As you can see I'm only doing `amd64`, `armv7`, and `arm64v8`.

**Comment by edwardreed81** [comment]
I know this is old, but has there been any change? I can't get this container to work on my Raspberry Pi 4 running Ubuntu Server 20.04 64bit.

**Comment by ryansch** [comment]
@edwardreed81 Try using `ryansch/unifi-rpi:5.12.72-arm64v8` instead of `latest`.

**Comment by edwardreed81** [comment]
Tried that earlier today (I know because I didn't have to pull that image just now) but I just tried it again and got almost no feedback:

```
Creating volume "unifi-2_config" with local driver
Creating volume "unifi-2_log" with local driver
Creating volume "unifi-2_log2" with local driver
Creating volume "unifi-2_run" with local driver
Creating volume "unifi-2_run2" with local driver
Creating volume "unifi-2_work" with local driver
Pulling unifi (ryansch/unifi-rpi:5.12.72-arm64v8)...
5.12.72-arm64v8: Pulling from ryansch/unifi-rpi
Digest: sha256:6fec56f4446025cd62095ca1e8cc98d995ac8c6ae82c9ddb5a24ad5f414ed4c8
Status: Downloaded newer image for ryansch/unifi-rpi:5.12.72-arm64v8
Creating unifi ... done
Attaching to unifi
```

If I do a `sudo netstat -tulnp` I can see that it opens ports 8880 and 8080, but never seems to get to the rest of them, including 8443.

I got the linuxserver.io image (https://github.com/linuxserver/docker-unifi-controller) working, so maybe they're doing something different?

**Comment by cklann1** [comment]
ryansch  - Thanks for your work!

Also trying to install on rp4/Ubuntu 20.04 server:
Linux rp4u 5.4.0-1011-raspi #11-Ubuntu SMP Fri May 8 07:43:33 UTC 2020 aarch64 aarch64 aarch64 GNU/Linux

Not able to connect to controller web interface 8443.

Seems to build container ok, but here's the server.log upon startup:
root@rp4u:/var/lib/unifi/logs# tail -f server.log
        at org.apache.coyote.AbstractProtocol.init(AbstractProtocol.java:581) ~[tomcat-embed-core-8.5.34.jar:8.5.34]
        at org.apache.coyote.http11.AbstractHttp11Protocol.init(AbstractHttp11Protocol.java:68) ~[tomcat-embed-core-8.5.34.jar:8.5.34]
        at org.apache.catalina.connector.Connector.initInternal(Connector.java:993) ~[tomcat-embed-core-8.5.34.jar:8.5.34]
        at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:107) ~[tomcat-embed-core-8.5.34.jar:8.5.34]
        ... 9 more
[2020-06-02T11:29:02,299] <localhost-startStop-1> INFO  system - ======================================================================
[2020-06-02T11:29:02,301] <localhost-startStop-1> INFO  system - UniFi 5.12.72 (build atag_5.12.72_13103 - release/release) is started
[2020-06-02T11:29:02,302] <localhost-startStop-1> INFO  system - ======================================================================
[2020-06-02T11:29:02,304] <localhost-startStop-1> INFO  system - BASE dir:/usr/lib/unifi
[2020-06-02T11:29:02,464] <localhost-startStop-1> INFO  system - Current System IP: 192.168.0.157
[2020-06-02T11:29:17,479] <localhost-startStop-1> INFO  system - Hostname: rp4u
[2020-06-02T11:29:18,875] <localhost-startStop-1> INFO  db     - waiting for db connection...
[2020-06-02T11:29:35,706] <db-server> INFO  db     - DbServer stopped
[2020-06-02T11:29:39,818] <db-server> INFO  db     - DbServer stopped
[2020-06-02T11:29:43,889] <localhost-startStop-1> INFO  db     - Connecting to mongodb://localhost:27117
[2020-06-02T11:29:43,942] <db-server> INFO  db     - DbServer stopped
[2020-06-02T11:29:48,054] <db-server> INFO  db     - DbServer stopped

Also tried linuxserver.io image (https://github.com/linuxserver/docker-unifi-controller), but couldn't get that to work either.

**Comment by ryansch** [comment]
From that output it almost seems like mongo isn't starting...

**Comment by cklann1** [comment]
Ack that...I've attached the mongod.log file - seems like it's having a problem opening a port?
2020-06-02T11:29:43.938+0000 I STORAGE  [initandlisten] exception in initAndListen: 0 assertion
[mongod.log](https://github.com/ryansch/docker-unifi-rpi/files/4720483/mongod.log)

Thanks!

**Comment by ryansch** [comment]
I have my rasp pi 4 with the latest ubuntu server and I've been able to reproduce this failure! I'll post updates here as I have time to work through this.

**Comment by cklann1** [comment]
Great - thanks, Ryan!  Although I'm kind of a newbie, I'd be happy to help in any way.

**Comment by jphellemons** [comment]
I also tried ryansch/unifi-rpi:5.12.72-arm64v8 and it did not came up (the webinterface) also on `Linux ubuntu 5.4.0-1012-raspi #12-Ubuntu SMP Wed May 27 04:08:35 UTC 2020 aarch64 aarch64 aarch64 GNU/Linux`

Description:    Ubuntu 20.04 LTS

**Comment by jphellemons** [comment]
I used a newer 5.14.9 and this is in the server.log:
```
[2020-06-28T07:17:56,967] <localhost-startStop-1> INFO  system - UniFi 5.14.9 (build atag_5.14.9_13726 - release/release) is started
[2020-06-28T07:17:56,968] <localhost-startStop-1> INFO  system - ======================================================================
[2020-06-28T07:17:56,970] <localhost-startStop-1> INFO  system - BASE dir:/usr/lib/unifi
[2020-06-28T07:17:57,005] <localhost-startStop-1> INFO  system - Current System IP: 192.168.0.112
[2020-06-28T07:17:57,028] <localhost-startStop-1> INFO  system - Hostname: UnknownHost
[2020-06-28T07:17:57,035] <localhost-startStop-1> INFO  system - ubic.env: prod
[2020-06-28T07:17:58,616] <localhost-startStop-1> INFO  db     - waiting for db connection...
[2020-06-28T07:18:00,118] <db-server> INFO  db     - DbServer stopped
[2020-06-28T07:18:00,119] <localhost-startStop-1> INFO  db     - Connecting to mongodb://localhost:27117
[2020-06-28T07:18:04,240] <db-server> INFO  db     - DbServer stopped
```

when I typ `mongo` I get this:
```
MongoDB shell version: 3.2.11
connecting to: test
2020-06-28T07:22:30.245+0000 W NETWORK  [thread1] Failed to connect to 127.0.0.1:27017, reason: errno:111 Connection refused
2020-06-28T07:22:30.246+0000 E QUERY    [thread1] Error: couldn't connect to server 127.0.0.1:27017, connection attempt failed :
connect@src/mongo/shell/mongo.js:229:14
@(connect):1:6

exception: connect failed
```

I used `ryansch/unifi-rpi:5.14.9-arm64v8` but there seems an issue with iptable because the mongo db cannot (re)start

**Comment by cklann1** [comment]
Interesting update...
I tried rebuilding container using docker-compose method, using your "stock" docker-compose.yml file, and it was showing the same symptoms / errors ([2020-06-18T12:36:08,797] <db-server> INFO  db     - DbServer stopped), but then jumped into unifi container's bash to speluk-around an look at error logs...wanted to use nano, so did an apt update, then apt install nano...dunno if it was coincidence, but that looked like it might've kickstarted the container - I was able to access unifi controller from 8443.
Here's a snippet of the unifi log file around that time:
```
[2020-06-18T12:36:00,579] <db-server> INFO  db     - DbServer stopped
[2020-06-18T12:36:04,686] <db-server> INFO  db     - DbServer stopped
[2020-06-18T12:36:08,797] <db-server> INFO  db     - DbServer stopped
[2020-07-03T13:07:29,272] <localhost-startStop-1> INFO  Version - HV000001: Hibernate Validator 6.1.2.Final
[2020-07-03T13:07:30,187] <localhost-startStop-1> INFO  system - ======================================================================
[2020-07-03T13:07:30,190] <localhost-startStop-1> INFO  system - UniFi 5.13.32 (build atag_5.13.32_13646 - release/release) is started
[2020-07-03T13:07:30,191] <localhost-startStop-1> INFO  system - ======================================================================
[2020-07-03T13:07:30,194] <localhost-startStop-1> INFO  system - BASE dir:/usr/lib/unifi
[2020-07-03T13:07:30,356] <localhost-startStop-1> INFO  system - Current System IP: 172.18.0.1
[2020-07-03T13:07:45,371] <localhost-startStop-1> INFO  system - Hostname: rp4u
[2020-07-03T13:07:45,380] <localhost-startStop-1> INFO  system - ubic.env: prod
[2020-07-03T13:07:46,543] <localhost-startStop-1> INFO  db     - waiting for db connection...
[2020-07-03T13:08:03,549] <localhost-startStop-1> INFO  db     - Connecting to mongodb://localhost:27117
[2020-07-03T13:08:07,984] <localhost-startStop-1> INFO  db     - Connecting to mongodb://localhost:27117
[2020-07-03T13:08:09,972] <localhost-startStop-1> INFO  productinfo - Using controller channel=RELEASE, firmware channel=RELEASE. Available controller channels=[RELEASE], available fi$
[2020-07-03T13:08:10,150] <localhost-startStop-1> WARN  dev    - failed to read /usr/lib/unifi/data/model_lifecycles.json - /usr/lib/unifi/data/model_lifecycles.json (No such file or $
[2020-07-03T13:08:11,160] <localhost-startStop-1> INFO  webrtc - WebRTC library version: EvoStream Media Server (www.evostream.com) build v2.9.1 - Gladiator - (built for Debian-9.2.1-$
[2020-07-03T13:08:13,468] <localhost-startStop-1> INFO  remote - Created directory: /usr/lib/unifi/logs/remote
[2020-07-03T13:08:14,882] <localhost-startStop-1> INFO  sdn    - Removed fingerbank token
[2020-07-03T13:07:30,191] <localhost-startStop-1> INFO  system - ======================================================================
[2020-07-03T13:07:30,194] <localhost-startStop-1> INFO  system - BASE dir:/usr/lib/unifi
[2020-07-03T13:07:30,356] <localhost-startStop-1> INFO  system - Current System IP: 172.18.0.1
[2020-07-03T13:07:45,371] <localhost-startStop-1> INFO  system - Hostname: rp4u
[2020-07-03T13:07:45,380] <localhost-startStop-1> INFO  system - ubic.env: prod
[2020-07-03T13:07:46,543] <localhost-startStop-1> INFO  db     - waiting for db connection...
[2020-07-03T13:08:03,549] <localhost-startStop-1> INFO  db     - Connecting to mongodb://localhost:27117
[2020-07-03T13:08:07,984] <localhost-startStop-1> INFO  db     - Connecting to mongodb://localhost:27117
[2020-07-03T13:08:09,972] <localhost-startStop-1> INFO  productinfo - Using controller channel=RELEASE, firmware channel=RELEASE. Available controller channels=[RELEASE], available fi$
[2020-07-03T13:08:10,150] <localhost-startStop-1> WARN  dev    - failed to read /usr/lib/unifi/data/model_lifecycles.json - /usr/lib/unifi/data/model_lifecycles.json (No such file or $
[2020-07-03T13:08:11,160] <localhost-startStop-1> INFO  webrtc - WebRTC library version: EvoStream Media Server (www.evostream.com) build v2.9.1 - Gladiator - (built for Debian-9.2.1-$
[2020-07-03T13:08:13,468] <localhost-startStop-1> INFO  remote - Created directory: /usr/lib/unifi/logs/remote
[2020-07-03T13:08:14,882] <localhost-startStop-1> INFO  sdn    - Removed fingerbank token
[2020-07-03T13:09:16,441] <autoupdate-check> INFO  productinfo - firmware[BZ2] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,446] <autoupdate-check> INFO  productinfo - firmware[BZ2LR] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,447] <autoupdate-check> INFO  productinfo - firmware[S216150] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,448] <autoupdate-check> INFO  productinfo - firmware[S224250] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,449] <autoupdate-check> INFO  productinfo - firmware[S224500] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,450] <autoupdate-check> INFO  productinfo - firmware[S248500] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,450] <autoupdate-check> INFO  productinfo - firmware[S248750] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,451] <autoupdate-check> INFO  productinfo - firmware[S28150] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,452] <autoupdate-check> INFO  productinfo - firmware[U2HSR] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,453] <autoupdate-check> INFO  productinfo - firmware[U2IW] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,453] <autoupdate-check> INFO  productinfo - firmware[U2Lv2] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,454] <autoupdate-check> INFO  productinfo - firmware[U2O] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,455] <autoupdate-check> INFO  productinfo - firmware[U2Sv2] new version (4.3.13.11253) is available
[2020-07-03T13:09:16,455] <autoupdate-check> INFO  productinfo - firmware[U5O] new version (4.3.13.11253) is available
 
```

**Comment by jphellemons** [comment]
> 
> Interesting update...
> I tried rebuilding container using docker-compose method, using your "stock" docker-compose.yml file, 
Can you share the yml file? what image did you use? `ryansch/unifi-rpi:5.14.9-arm64v8` ?

**Comment by cklann1** [comment]
Used ryansch's docker-compose.yml - using the :latest tag.
Also, had sudo apt update && apt dist-upgraded the rpi4 host Ubuntu OS a
couple days ago...wonder if the driver updates did something to help?

**Comment by jphellemons** [comment]
Thank you @cklann1 But I got 

>Bad Request
This combination of host and port requires TLS.

when trying to access port 8443 (or 8843) 
my portscanner (angry ip) does show those ports

It does not forward http to https. So if someone has the same message: use https and port 8443

**Comment by lucaciraolo** [comment]
I'm trying this image on Manjaro for rpi4 and can't get the web interface to load either. Any help would be much appreciated!

**Comment by ryansch** [comment]
I took a run at this last weekend and tried upgrading us to newer mongo and debian buster. It's a bit of a mess because the controller wants mongo <= 3.6 (and that's just for 5.13.10 and newer) and java 8. Those are both old enough that it's becoming hard to get packages/container images for them.

I may have to break down and package both of those dependencies so we have more control.

**Comment by ryansch** [comment]
Note: I'm trying to get us on mongo 3.6 and debian buster for the new 6.x controller release.

**Comment by jphellemons** [comment]
I uncommented the java part. this is my docker-compose now:

```
version: '2'
services:
  unifi:
    image: ryansch/unifi-rpi:latest
    container_name: unifi
    restart: always
    network_mode: host
    # Uncomment the following to set java options
    environment:
       JAVA_OPTS: -Xmx512M
    volumes:
      # Unifi v5.0.7 creates all of these directories (some remain empty)
      - config:/var/lib/unifi
      - log:/usr/lib/unifi/logs
      - log2:/var/log/unifi
      - run:/usr/lib/unifi/run
      - run2:/run/unifi
      - work:/usr/lib/unifi/work
volumes:
  config:
    driver: local
  log:
    driver: local
  log2:
    driver: local
```

it is available at `https://my-lan-ip:8443`

I thought that I would have the 6.0 version because I have latest in the docker compose file and I did this:
`docker-compose up -d --no-deps --build unifi`
as found on https://staxmanade.com/2016/09/how-to-update-a-single-running-docker-compose-container/

UI 5.13.32.0
Backend 5.13.32
Build atag_5.13.32_13646

**Comment by ryansch** [comment]
I accidentally kicked the power brick loose for my test pi for this issue. I started from scratch and installed ubuntu 20.04.1 arm64, docker, docker-compose, and unifi controller 5.13.32. It booted right up without any of the weirdness I ran into last time I tried this.

I did nothing and now it's fixed! 🎉

**Comment by ryansch** [comment]
@jphellemons 6.x is still in RC. I won't mark it as stable until upstream does.

**Comment by ryansch** [comment]
Note that we're still on debian stretch for now.

**Comment by jphellemons** [comment]
> @jphellemons 6.x is still in RC. I won't mark it as stable until upstream does.

Awesome that you are preparing for 6.0.
Thank you for the support and nice docker images 👍

**Comment by cooldil** [comment]
Pi 4 8GB running 64-bit PiOS (Rasbian)

I had to map /work to the config volume in order for the container to start up.  Until I did so, the container to stop after complaining about unable to find the /work/system.properties file (which is technically in the /var/lib/unifi folder which is mapped to the config volume).

Hope this helps anyone else that might be having the same issue.

For the record, I am running with just TWO volumes (but three mapped paths).

   config:/var/lib/unifi
   config:/work
   work:/usr/lib/unifi

Now if only I can figure out how not to run the container as the root (uid 0).....
